R/epiparam.R

```diff
@@ -0,0 +1,263 @@
+    tolower(params$author),
+    method = "radix"
+  ), ]
+
+  if (epi_dist != "all") {
+    # filter by delay distribution
+    params <- params[params$epi_distribution == epi_dist, ]
+  }
+
+  # reset indexing of rows
+  rownames(params) <- NULL
+
+  structure(params, class = c("epiparam", "data.frame"))
+
+}
+
+#' Create an `epiparam` object
+#'
+#' @param epi_dist A character string of which epidemiological distributions
+#' to select
+#'
+#' @return `epiparam` object
+#' @export
+#'
+#' @examples
+#' # the object can be made without arguments
+#' eparam <- epiparam()
+#'
+#' # specifying incubation periods
+#' incub_eparam <- epiparam("incubation")
+epiparam <- function(epi_dist = c("all",
+                                  "incubation_period",
+                                  "onset_to_admission",
+                                  "onset_to_death",
+                                  "serial_interval",
+                                  "generation_time",
+                                  "offspring_distribution")) {
+
+  # check input
+  epi_dist <- match.arg(arg = epi_dist, several.ok = FALSE)
+
+  # create epiparam object
+  epiparam <- new_epiparam(epi_dist = epi_dist)
+
+  # validate epiparam object
+  validate_epiparam(epiparam)
+
+  # return epiparam object
+  epiparam
+}
+
+#' `epiparam` class validator
+#'
+#' @param epiparam An `epiparam` object
+#'
+#' @return Nothing, errors when invalid `epiparam` object is provided
+validate_epiparam <- function(epiparam) {
+
+  if (!inherits(epiparam, "epiparam")) {
+    stop("Object should be of class epiparam")
+  }
+
+  # check for class invariants
+  stopifnot(
+    "epiparam object does not contain the correct columns" =
+      c("disease", "epi_distribution", "author", "year", "vector_borne",
+        "extrinsic", "prob_distribution", "discretised", "censorred",
+        "right_truncated", "phase_bias_adjusted", "DOI") %in%
+      colnames(epiparam),
+    "disease needs to be a character" =
+      is.character(epiparam$disease),
+    "epi_distribution needs to be a character" =
+      is.character(epiparam$epi_distribution),
+    "author must be a string" =
+      is.character(epiparam$author),
+    "year must be a numeric" =
+      is.numeric(epiparam$year),
+    "vector_borne must be a logical" =
+      is.logical(epiparam$vector_borne),
+    "extrinsic must be a logical" =
+      is.logical(epiparam$extrinsic),
+    "prob_distribution needs to be a character" =
+      is.character(epiparam$prob_distribution),
+    "discretised needs to be a logical" =
+      is.logical(epiparam$discretised),
+    "censorred needs to be a logical" =
+      is.logical(epiparam$censorred),
+    "right_truncated needs to be a logical" =
+      is.logical(epiparam$right_truncated),
+    "phase_biase_adjusted needs to be a logical" =
+      is.logical(epiparam$phase_bias_adjusted),
+    "DOI needs to be a character" =
+      is.character(epiparam$DOI)
+  )
+
+  invisible(epiparam)
+}
```

Comment:
1. Suggest expanding description to explain what the `epiparam` class is about and what a typical use case is.
	1. Am I right in assuming that it is primarily a way to access the data stored in the package? 
	2. Should it be possible to create a local and temporary working library of delay distributions, i.e., make a number of `epidist` objects and build an `epiparam` library from them? This could be useful functionality that adds to `bind_epiparam` for users building their own parameter libraries, for example, to prepare for contributing to `{epiparameter}`.
2. On the whole I liked the ability to make `epidist` objects from `epiparam` dataframes - however, the function `as_epidist()` was difficult to find without a vignette or Readme. Before I found it, I essentially suggested adding a similar function. I had also sketched out a workflow for these conversions, which I'm leaving here anyway in case it's useful, say for a vignette: *(a)* Read in the parameter library using `epiparam()`, *(b)* Subset - while reading or after - the library for a disease and a distribution, *(c)* Make `epidist` objects from each row of the `epiparam`, and potentially *(d)* build a workflow that takes these `epidists`. For example, when dealing with a new outbreak, get all onset to death distributions for that disease-causing pathogen from the library, make each an `epidist`, and run separate CFR calculations using each one - this basically gets at the question: what is the CFR if this new outbreak is caused by pathogen similar to one of several previously known ones?
3. Is there a reason that some disease name are capitalised while others are not? See especially diseases with proper nouns, such as "west nile fever" or "ebola virus disease" vs. "Hantavirus pulmonary syndrome" or "Rift Valley fever". See also naming of place-virus disease names, such as "Hantavirus" (no space) vs. "ebola virus" (with space).

Reply:
Following on from 1, could be helpful to have example of how to interact simply with database in the documentation, e.g. 
```
db <- epiparam()
db[db$disease == "smallpox"]
```
(or dplyr equivalent)

Reply:
Resolved in 71187f72ab3fbf6a9e77e8841b6892e871990627

Reply:
Thanks for the example workflow. I will provide something similar in a vignette. I will also include examples of handling the {epiparameter} classes in a vignette using both base and tidyverse examples (noted here #94)